client/blocks/importer/wordpress/upgrade-plan/test/index.tsx

```diff
@@ -0,0 +1,206 @@
+	};
+} );
+
+function renderUpgradePlanComponent( props: ComponentPropsWithoutRef< typeof UpgradePlan > ) {
+	const queryClient = new QueryClient();
+
+	return renderWithProvider(
+		<QueryClientProvider client={ queryClient }>
+			<UpgradePlan { ...props } />
+		</QueryClientProvider>,
+		{
+			initialState: {},
+			reducers: {},
+		}
+	);
+}
+
+const DEFAULT_SITE_ID = 123;
+const DEFAULT_SITE_SLUG = 'test-example.wordpress.com';
+
+const DEFAULT_SITE_CAPABILITIES = {
+	activate_plugins: true,
+	activate_wordads: true,
+	delete_others_posts: true,
+	delete_posts: true,
+	delete_users: true,
+	edit_others_pages: true,
+	edit_others_posts: true,
+	edit_pages: true,
+	edit_posts: true,
+	edit_theme_options: true,
+	edit_users: true,
+	list_users: true,
+	manage_categories: true,
+	manage_options: true,
+	moderate_comments: true,
+	own_site: true,
+	promote_users: true,
+	publish_posts: true,
+	remove_users: true,
+	upload_files: true,
+	view_hosting: true,
+	view_stats: true,
+};
+
+function getUpgradePlanProps(
+	customProps: Partial< ComponentPropsWithoutRef< typeof UpgradePlan > > = {}
+): ComponentPropsWithoutRef< typeof UpgradePlan > {
+	const defaultProps: ComponentPropsWithoutRef< typeof UpgradePlan > = {
+		site: {
+			ID: DEFAULT_SITE_ID,
+			URL: 'https://test-example.wordpress.com',
+			capabilities: DEFAULT_SITE_CAPABILITIES,
+			description: 'test site',
+			domain: 'test-example.wordpress.com',
+			jetpack: false,
+			launch_status: 'launched',
+			locale: 'en',
+			logo: { id: 'logo', sizes: [ '100x100' ], url: 'https://test-example.wordpress.com/logo' },
+			name: 'Test Site',
+			slug: DEFAULT_SITE_SLUG,
+			title: 'Test Site',
+		},
+		isBusy: false,
+		ctaText: '',
+		navigateToVerifyEmailStep: () => {},
+		onCtaClick: () => {},
+	};
+
+	return {
+		...defaultProps,
+		...customProps,
+	};
+}
+
+const mockApi = () => nock( 'https://public-api.wordpress.com:443' );
+
+const API_RESPONSE_ELIGIBLE = {
+	eligible: true,
+};
+
+const API_RESPONSE_INELIGIBLE_UNVERIFIED_EMAIL = {
+	eligible: true,
+	error_code: 'email-unverified',
+};
+
+describe( 'UpgradePlan', () => {
+	beforeAll( () => nock.disableNetConnect() );
+
+	it( 'should render the default Continue CTA and trigger onCtaClick when no CTA is supplied', async () => {
+		const mockOnCtaClick = jest.fn();
+
+		renderUpgradePlanComponent( getUpgradePlanProps( { onCtaClick: mockOnCtaClick } ) );
+
+		await userEvent.click( screen.getByRole( 'button', { name: /Continue/ } ) );
+
+		await waitFor( () => expect( mockOnCtaClick ).toHaveBeenCalled() );
+	} );
+
+	it( 'should render a custom CTA admd triggerOnCtaClick when ctaText is supplied', async () => {
+		const mockOnCtaClick = jest.fn();
+
+		renderUpgradePlanComponent(
+			getUpgradePlanProps( { ctaText: 'My Custom CTA', onCtaClick: mockOnCtaClick } )
+		);
+
+		await userEvent.click( screen.getByRole( 'button', { name: /My Custom CTA/ } ) );
+
+		await waitFor( () => expect( mockOnCtaClick ).toHaveBeenCalled() );
```

Comment:
If we only want to see if the was rendered with the new label you can use:
```suggestion
		expect( await screen.findByRole( 'button', { name: /My Custom CTA/ } )).toBeVisible();
```
And we don't need to test again the onCtaClick except we have some variation based on the CTA

Reply:
I prefer making double sure that things which should touch the button still work. 😁 So I will leave this as-is.